lib/AST/ASTBuilder.cpp

```diff
@@ -178,7 +183,7 @@ clang::StringLiteral *ASTBuilder::CreateStrLit(std::string val) {
 clang::FloatingLiteral *ASTBuilder::CreateFPLit(llvm::APFloat val) {
   auto size{llvm::APFloat::getSizeInBits(val.getSemantics())};
   auto type{GetLeastRealTypeForBitWidth(size)};
-  CHECK(!type.isNull()) << "Unable to infer type for given value.";
+  { CHECK_THROW(!type.isNull()) << "Unable to infer type for given value."; }
```

Comment:
I think if we wanted to avoid the `{` then you could maybe do something like:

```
template <typename E>
class ThrowerStream {
  std::stringstream &ss;
  bool can_throw{true};
  inline explicit ThrowerStream(std::stringstream &ss_=std::stringstream())
    : ss(ss_) {}

  ~ThrowerStream(void) {
    if (can_throw) {
      throw E(ss.str());
    }
  }

  template <typename T>
  ThrowerStream operator<<(T &&val) {
    can_throw = false;
    ss << std::forward<T>(val);
    return ThrowerStream(ss);
  }
};
```

Reply:
I'm loving these tricks :)